posthog/constants.py

```diff
@@ -89,6 +89,14 @@
 FUNNEL_FROM_STEP = "funnel_from_step"
 FUNNEL_TO_STEP = "funnel_to_step"
 FUNNEL_STEP = "funnel_step"
+FUNNEL_TYPE = "funnel_type"
+
+
+class FunnelType(str, Enum):
+    STRICT = "strict"
+    UNORDERED = "unordered"
+    TRENDS = "trends"
```

Comment:
Ordering vs. viz type are kind of orthogonal actually – you can have steps, trends, or time to convert graphs, and each of them can work with strict ordering AND with any ordering. I'm pretty sure we will allow mixing and matching this, as do other analytics tools. So I suggest to split this into two enums: `FunnelOrdering` and `FunnelVizType`.

BTW, is "strict" the best notion for what we have now? For Amplitude has a concept of "exact" order, meaning events must happen exactly one after another to count (none in between). This doesn't seem ideal for us (especially considering we do autocapture), but also seems like our way, while "ordered", is not particularly "strict". Feel free to disregard me rambling about this detail though. :D

Reply:
This makes *A lot* of sense! Didn't realise Trends was just a visualisation type.